sdk/ml/azure-ai-ml/azure/ai/ml/_schema/job/input_output_entry.py

```diff
@@ -35,6 +35,8 @@ def generate_path_property(azureml_type):
     return UnionField(
         [
             ArmVersionedStr(azureml_type=azureml_type),
+            ArmVersionedStr(azureml_type=AzureMLResourceType.DATASTORE),
```

Comment:
I believe these are the patterns we need to support. https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/ml/azure-ai-ml/azure/ai/ml/constants/_common.py#:~:text=JOB_URI_FORMAT%20%3D%20%22azureml,azureml%3A//datastores/(%5B%5E/%5D%2B)/paths/(.%2B)%22

AzureML also has public docs on this, but it's not comprehensive unfortunately. https://learn.microsoft.com/en-us/azure/machine-learning/how-to-read-write-data-v2?tabs=cli#supported-paths

Reply:
Can you also add some unit tests? Thx.

Reply:
Given this is an issue only raised by the VSCode extension, I believe we may be better off leaving testing to their end.